### DISCUSSION TITLE
Trying Serverpod, server worked till adding auth module

### DISCUSSION THREAD

**Discussion by Ekkas** [serverpod/serverpod#689] [Q&A] [answered]
Hello everyone,

I recently installed the latest version of ServerPod, version 1.0.1, and started a new project. I tested the server and the Flutter app and both ran without any issues. However, when I tried to add the auth module following [the provided instructions](https://docs.serverpod.dev/concepts/modules), I ran into an error.

First, I edited the 'generator.yaml' file and added the following lines:
modules:
  serverpod_auth:
    nickname: auth

Next, I edited the 'pubspec.yaml' file and added the following line under dependencies:
serverpod_auth_server: ^1.0.1

I also made similar changes to the 'pubspec.yaml' files in my client and Flutter apps as per the instructions.

I ran the SQL generation and confirmed that the SQL tables were created. I then restarted the container and ran 'serverpod generate'. I even rebooted my Windows PC, but the error persisted. Whenever I run 'dart .\bin\main.dart' in the server, I get the following error:

/C:/Users/anton/AppData/Local/Pub/Cache/hosted/pub.dev/sortedmap-0.5.2/lib/src/treeset.dart:17:3: Error: Type 'BidirectionalIterator' not found.
BidirectionalIterator<V> fromIterator(V anchor,

I'm not sure what the issue is or how to fix it. Has anyone else experienced this issue or have any suggestions on how to resolve it?

Thank you in advance for your help.
Anton

**Comment by Pingear** [comment]
Have you seen this video? - https://www.youtube.com/watch?v=y7oLNlbWbUY Following it, I managed to add an authorization module, after which it became easier to navigate in the documentation itself.

**Comment by yahu1031** [comment, marked as answer]
`BidirectionalIterator` was deprecated after 3.0.0 dart version. Related issue - https://github.com/appsup-dart/sortedmap/issues/5

Update: https://api.dart.dev/stable/2.19.2/dart-core/BidirectionalIterator-class.html

**Reply by Ekkas** [reply]
Thank you yahu1031
I have versions:
Flutter 3.8.0-15.0.pre.4 • channel master • https://github.com/flutter/flutter.git
Framework • revision 5ccb4d4b95 (17 minutes ago) • 2023-02-22 13:32:40 -0800
Engine • revision 3aa112c75c
Tools • Dart 3.0.0 (build 3.0.0-261.0.dev) • DevTools 2.21.1

Should I try to downgrade Dart to pre 3.0.0 or is there maybe another solution?
Thanks again.

**Reply by yahu1031** [reply]
Use flutter stable version. Or else it will break if dart is 3.0.0

**Reply by Ekkas** [reply]
Thank you that solved the problem.

**Reply by yahu1031** [reply]
@Ekkas now overriding `sortedmap` with latest version in the whole project must ideally work.

CC: @vlidholt - Serverpod ready for Dart 3
